akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -435,16 +436,30 @@ object Behavior {
       val b2 = Behavior.start(b, ctx)
       if (!Behavior.isAlive(b2) || !messages.hasNext) b2
       else {
-        val nextB = messages.next() match {
+        val next = messages.next()
+        val nextB = try next match {
           case sig: Signal ⇒ Behavior.interpretSignal(b2, ctx, sig)
           case msg         ⇒ Behavior.interpretMessage(b2, ctx, msg)
+        } catch {
+          case NonFatal(e) ⇒
+            Behavior.failed[T](InterpretBehaviorException[T](e, b2, next))
```

Comment:
The thinking here is capture the exact
1. original exception
2. failed behavior
3. message processed that raises it

Is there a better way?

Reply:
Looking at this makes me wonder if `interpretMessages` really belongs as a general operation in `Behavior` - it is only used from stash plus now we need special logic in it for stash.

Reply:
> now we need special logic in it for stash.

Well, the way I wrote this specific part was trying to be unrelated to stash, applicable to anything. It's just catching an exception from any caller wanting `Behavior.interpretMessage` or `Behavior.interpretSignal` and wrapping a potential error for more error context.

Reply:
Unless you mean `interpretMessage` belongs more to `StashBuffer` ops, which if true, yes perhaps it doesn't belong there.